reframe/core/schedulers/slurm.py

```diff
@@ -611,13 +611,9 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(self.name)
 
-    def is_available(self):
-        return all([self._states == {'IDLE'}, self._partitions,
-                    self._active_features, self._states])
-
-    def under_maintenance(self):
-        return all([self._states == {'MAINT'}, self._partitions,
-                    self._active_features, self._states])
+    def in_state(self, state):
+        return all([self._states == set(state.upper().split('+')),
```

Comment:
```suggestion
        return all([self._states >= set(state.upper().split('+')),
```
A node in `MAINT+DRAIN` is both in `MAINT` and `DRAIN` states, so both queries should return true.

Reply:
I was thinking of this as well. The problem is that then you lose the option of asking for nodes that satisfy both states.

Reply:
Can you give an example?

Reply:
If you want to get the nodes that are both in `MAINT` and `DRAIN` then you cannot do it since according to the suggestion you will get also the ones that are either only  `MAINT` or `DRAIN`.

Reply:
Are you sure? Wouldn't `self._states` be `{'MAINT'}` or `{'DRAIN'}` in that case and what the user gave wouldn't be `{'MAINT', 'DRAIN'}`, so the `>=` will return false?

Reply:
You are right, my bad. I will implement this.